### ISSUE TITLE
Bracket notation not working

### ISSUE THREAD

**Issue by mrleblanc101** [kazupon/vue-i18n#1100]
Hi,
I have some dynamic data from my API which I need to translate use Vue-i18n.
So I created an object which hold all the values and I wanted to use the dynamic data from my API as the object keys.
Se bellow:
```
    type: {
        activite: 'Activity',
        capsule: 'Capsule',
        programme: 'Program',
        parcours: 'Path',
    },
    ```

Since the translation key is dynamic, I expected to use the bracket notation like in Vanilla JS ``{{ $t(`formation.type[${formation.approche_type}]`) }}`` but this didn't work.

After some trial and error, I realised I could use the dot notation even with a dynamic key ``{{ $t(`formation.type.${formation.approche_type}`) }}``. 
This would normally not be valid in Javascript.
I wonder if there is a reason why the bracket notation is not support and if this could be a feature request.

### vue & vue-i18n version
Vue v2.6.12
Nuxt 2.14.12
Nuxt-i18n 6.16.0
Vue-i18n 8.18.1

**NOTE: vue-i18n does not support Vue.js 3.0 yet.**
**Check https://github.com/intlify/vue-i18n-next**

**Comment by mrleblanc101** [comment]
Seem to work but raise warning ` WARN  [vue-i18n] Value of key 'formation.type.activite' is not a string or function !` even if the translation exist as you can see above.

**Comment by exoego** [comment]
vue-i18n's key syntax is very similar to JS notation, but not the subset/superset of JS.
IMHO, bracket notation is good to have, but not high priority since there are other notations for same use case.

**Comment by mrleblanc101** [comment]
@exoego Maybe the doc should make this clearer ? I don't think this is explained anywhere.

I know that $t() accept a string so I should not necessarily expect it to behave like an object, but still.

Also, any clue about the warnings ? I had to enable `silentWarnFallback: true` to get rid of them. Everything seems to be working fine so I'm not sure why they are appearing.

**Comment by exoego** [comment]
> Value of key 'formation.type.capsule' is not a string or function !

These warning are shown when the specified key does not have the corresponding item of string/function.

The above example does not have `formation` field, so `formation.type.capsule` does not have the corresponding item.

```javascript
messages: {
    fr: {
        type: {
            activite: 'Activité',
            capsule: 'Capsule',
```

**Comment by exoego** [comment]
> Maybe the doc should make this clearer ? I don't think this is explained anywhere.

This sounds nice 👍

**Comment by mrleblanc101** [comment]
@exoego I might have oversimplified my example. It does have this sub object. I think the issue might just be that vue-i18n does not like dynamic keys

**Comment by exoego** [comment]
Oh, I see. so `formation.type.capsule` is valid key.

> vue-i18n does not like dynamic keys

Yes, this is correct.
Keys passed to vue-i18n should be static.
If keys are dynamic depending on user input or something, it is allowed to dynamically create a static key like you explained:

```js
$t(`foo.bar.${someValue}`); // string interpolation, or template literal
$t('foo.bar.' + someValue); // concatenation
```

**Comment by mrleblanc101** [comment]
@exoego Yes, I'm using `$t(`foo.bar.${someValue}`);` because otherwise it doesn't work. But I'm still getting the warning, but it works.

**Comment by lucasferreiralimax** [comment]
Template string of javascript work good for me.
{{ $t(`formation.type.${formation.approche_type}`) }}

**Comment by mrleblanc101** [comment]
@lucasferreiralimax Clearly you did not read very carefully
